lib/core/api.js

```diff
@@ -161,12 +161,12 @@ api.user.updateProfile = function(userId, updates, cb){
     if(err){
       return cb(err);
     }
-    var patient = _.get(profile, 'patient', {});
-    var patient_updates = _.get(updates, 'patient', {});
-    var new_patient = _.assign(patient, patient_updates);
-    var new_profile = _.merge(profile, updates);
-    new_profile.patient = new_patient;
-    return api.user.addProfile(userId, new_profile, cb);
+    var newProfile = _.merge(profile, updates, function(original, update){
```

Comment:
This is a place where you've missed just using the factored out `actionUtils.mergeProfileUpdates`, right? Or is there a subtle difference I'm missing?